src/rider/main/kotlin/me/fornever/avaloniarider/previewer/AvaloniaPreviewerSessionController.kt

```diff
@@ -198,30 +230,85 @@ class AvaloniaPreviewerSessionController(
     ).apply {
         sessionStarted.advise(lifetime) {
             statusProperty.value = Status.Working
+            pendingRestart = false
+            updateXamlResultProperty.set(UpdateXamlResultMessage())
 
             sendClientSupportedPixelFormat()
 
+            var document: Document? = null
             application.runReadAction {
-                val document = FileDocumentManager.getInstance().getDocument(xamlFile)!!
+                document = FileDocumentManager.getInstance().getDocument(xamlFile)
+            }
+            if (document == null) {
+                logger.warn("Unable to obtain document for $xamlFile")
+                return@advise
+            }
+            val currentDocument = document!!
+
+            val pathRetryDisposable = lifetime.createNestedDisposable(
+                "AvaloniaPreviewerSessionController.projectPathRetry"
+            )
+            val pathRetryAlarm = Alarm(pathRetryDisposable)
```

Comment:
Not sure I understand this logic. Let's get rid of `Alarm` (and replace with some more direct coroutined code, like a loop that tries to calculate what's needed) and iterate on this again.